### PR TITLE
[ORA-680] Query Server methods for getting mempool state from CLI+RPC

### PR DESCRIPTION
**Pull request by relyt29**
It was a low hanging fruit 1 pointer ticket so I thought I'd knock it out...

### CODE REVIEW

File: x/emissions/errors.go [hunk 1]
```diff
@@ -55,16 +55,6 @@ var (
 	ErrNotInTopicCreationWhitelist                       = errors.Register(ModuleName, 50, "not in topic creation whitelist")
 	ErrNotInWeightSettingWhitelist                       = errors.Register(ModuleName, 51, "not in topic weight setting whitelist")
 	ErrTopicNotEnoughDemand                              = errors.Register(ModuleName, 52, "topic not enough demand")
-	ErrSetParamsVersion                                  = errors.Register(ModuleName, 53, "Error Setting Params: Version")
-	ErrSetParamsEpochLength                              = errors.Register(ModuleName, 54, "Error Setting Params: EpochLength")
-	ErrSetParamsEmissionsPerEpoch                        = errors.Register(ModuleName, 55, "Error Setting Params: EmissionsPerEpoch")
-	ErrSetParamsMinTopicUnmetDemand                      = errors.Register(ModuleName, 56, "Error Setting Params: MinTopicUnmetDemand")
-	ErrSetParamsMaxTopicsPerBlock                        = errors.Register(ModuleName, 57, "Error Setting Params: MaxTopicsPerBlock")
```
Comment (kpeluso) [inline x/emissions/errors.go:62]: why are these being removed? no longer used?

File: x/emissions/keeper/query_server.go [hunk 2]
```diff
@@ -226,3 +226,54 @@ func (qs queryServer) GetRegisteredTopicsIds(ctx context.Context, req *state.Que
 
 	return &state.QueryRegisteredTopicsIdsResponse{TopicsIds: topicsIds}, nil
 }
+
+func (qs queryServer) GetInferenceRequest(ctx context.Context, req *state.QueryInferenceRequestRequest) (*state.QueryInferenceRequestResponse, error) {
+	valid := state.IsValidRequestId(req.RequestId)
```
Comment (kpeluso) [inline x/emissions/keeper/query_server.go:231]: surprised we need validation here... guess it doesn't hurt.

**Review by kpeluso — COMMENTED** [review]
not explicit approval just yet til ik what's up w the errors

File: x/emissions/errors.go (hunk 1 above)
Comment (relyt29) [reply x/emissions/errors.go:62]: spoke about this over slack

File: x/emissions/keeper/query_server.go (hunk 2 above)
Comment (relyt29) [reply x/emissions/keeper/query_server.go:231]: we don't actually, we could just make the request and then do an if statement on collections.IsNotFound but I think this is more defense in depthable code and it's probably about the same for performance

File: x/emissions/keeper/query_server.go [hunk 3]
```diff
@@ -226,3 +226,54 @@ func (qs queryServer) GetRegisteredTopicsIds(ctx context.Context, req *state.Que
...
+		return nil, err
+	}
+	demandLeft, err := qs.k.GetRequestDemand(ctx, req.RequestId)
+	if err != nil {
+		return nil, err
+	}
+	return &state.QueryInferenceRequestResponse{InferenceRequest: &inferenceRequest, DemandLeft: demandLeft}, nil
+}
+
+func (qs queryServer) GetAllInferenceRequests(ctx context.Context, req *state.QueryAllInferenceRequestsRequest) (*state.QueryAllInferenceRequestsResponse, error) {
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/keeper/query_server.go:253]: sure will rename

File: x/emissions/module/autocli.go [hunk 4]
```diff
@@ -93,6 +93,28 @@ func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
...
+					Short:     "Get Inference Request in mempool by topic id and request id",
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "topic_id"},
+						{ProtoField: "request_id"},
+					},
+				},
+				{
+					RpcMethod: "GetAllInferenceRequests",
+					Use:       "get-all-inference-requests",
+					Short:     "Get All Inference Requests in mempool",
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/module/autocli.go:108]: yes will add

File: x/emissions/module/autocli.go [hunk 5]
```diff
@@ -93,6 +93,28 @@ func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
 						{ProtoField: "libp2p_key"},
 					},
 				},
+				{
+					RpcMethod: "GetInferenceRequest",
+					Use:       "get-inference-request [topic_id] [request_id]",
+					Short:     "Get Inference Request in mempool by topic id and request id",
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/module/autocli.go:99]: yes will add

**Review by kpeluso — APPROVED** [review]

**Pull request merged**